vignettes_src/orderly.Rmd

```diff
@@ -0,0 +1,104 @@
+---
+title: "Using didehpc to run orderly tasks"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+    toc_depth: 2
+vignette: >
+  %\VignetteIndexEntry{Using didehpc to run orderly tasks}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+<!-- HEADER -->
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  error = FALSE,
+  comment = "#>"
+)
+orderly_root <- orderly:::prepare_orderly_example("demo")
+```
+
+When using [orderly](https://github.com/vimc/orderly) some tasks might be slow to run and would be better suited to running on a remote machine. You can use the cluster and [orderly bundles](https://www.vaccineimpact.org/orderly/articles/bundles.html) to achieve this. If your orderly instance has an [OrderlyWeb](https://github.com/vimc/orderly-web) remote then you should favour running tasks on the configured remote via `orderly::orderly_run_remote` (unless it is too slow even for the OrderlyWeb remote). If your orderly instance is using a [sharepoint remote](https://github.com/vimc/orderly.sharepoint) or has no remote then you can use the cluster to run your slow running tasks.
+
+## Running a task on the cluster
+
+To run a single task or report on the cluster start by initialising your bundle on a network drive. For example on Linux:
+
+```{r}
+root <- "~/net/home/contexts"
+path_bundles <- file.path(root, "bundles")
+
+## Create bundle
+bundle <- orderly::orderly_bundle_pack(path_bundles, 
+                                       "minimal", 
+                                       root = orderly_root)
+```
+
+Then setup the cluster with required packages, you will need `orderly` at a minimum. You can use the list of packages from the `orderly.yml` e.g. for report `internal-2017-population-TUV-MHL`
+
+```{r}
+orderly_packages <- yaml::read_yaml(
+  file.path(orderly_root, "src/minimal/orderly.yml"))$packages
+packages <- list(loaded = c("orderly", orderly_packages))
+config <- didehpc::didehpc_config(workdir = root)
+ctx <- context::context_save(root, packages = packages)
+obj <- didehpc::queue_didehpc(ctx, config = config)
+```
+
+The orderly task can then be run via `orderly::orderly_bundle_run`, being careful to make sure the paths are relative to the `workdir` passed in `didehpc_config`
+
+```{r}
+path <- file.path("bundles", basename(bundle$path))
+output_path <- "output"
+t <- obj$enqueue(orderly::orderly_bundle_run(path, output_path))
+```
+
+When the job has completed you can import the returned bundle. If you are on windows then the `path` in the result should just work. If you are on Mac or Linux you will need to construct the path
+
+```{r}
+output <- strsplit(t$wait(100)$path, "\\\\")[[1]]
```

Comment:
I'd split this to have

````
Then we need to wait for the bundle to complete.

```{r}
t$wait(100)
```
````

Now that it has completed, we can import it...

Reply:
However, we should never need to compute on the filename like this. Are you looking for `basename()` here perhaps?

Reply:
So `basename` doesn't work because of how it returns the path. It gives us something like `Q:\\path\\to\\file` which `basename` doesn't change
```
> x <- "Q:\\path\\to\\file"
> basename(x)
[1] "Q:\\path\\to\\file"
```
I'm guessing `basename` might work on windows though?

Reply:
Oh they can just use the path as is on windows